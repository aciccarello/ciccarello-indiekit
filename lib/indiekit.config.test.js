import { jest } from "@jest/globals";
import config from "../indiekit.config";

beforeAll(() => {
  jest.spyOn(console, "log").mockReturnValue();
});

afterAll(() => {
  jest.restoreAllMocks();
});

it("should be able to import the config without failing", async () => {
  expect(config).toMatchSnapshot();
});
