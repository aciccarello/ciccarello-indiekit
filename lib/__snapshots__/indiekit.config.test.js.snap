// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to import the config without failing 1`] = `
{
  "@indiekit/endpoint-media": {
    "imageProcessing": {
      "resize": {
        "fit": "inside",
        "width": 2048,
        "withoutEnlargement": true,
      },
    },
  },
  "@indiekit/store-file-system": {
    "directory": "../ciccarello.me",
    "name": "@indiekit/store-file-system",
  },
  "application": {
    "mongodbUrl": "mongodb://127.0.0.1:27017/?directConnection=true&serverSelectionTimeoutMS=2000&appName=indiekit",
    "themeColor": "#af1e0b",
  },
  "plugins": [
    "@indiekit/store-file-system",
    "@indiekit/endpoint-media",
  ],
  "publication": {
    "categories": "http://localhost:8080/posts/tags/index.json",
    "enrichPostData": true,
    "me": "http://localhost:8080",
    "postTemplate": [Function],
    "postTypes": {
      "article": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Blog",
        "post": {
          "path": "posts/blog/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "blog/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "bookmark": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Bookmark",
        "post": {
          "path": "posts/links/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "like": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Like",
        "post": {
          "path": "posts/links/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "note": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Note",
        "post": {
          "path": "posts/notes/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "photo": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Photo",
        "post": {
          "path": "posts/photos/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "recipe": {
        "fields": {
          "category": {},
          "content": {
            "required": true,
          },
          "date": {
            "required": true,
          },
          "name": {
            "required": true,
          },
          "photo": {},
          "visibility": {},
        },
        "h": "recipe",
        "media": {
          "path": "assets/img/recipe-ik-{filename}",
        },
        "name": "Recipe",
        "post": {
          "path": "posts/recipes/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
      "reply": {
        "media": {
          "path": "assets/img/{yyyy}-{DDD}-{filename}",
        },
        "name": "Reply",
        "post": {
          "path": "posts/replies/{yyyy}-{MM}-{dd}-{slug}.md",
          "url": "posts/{yyyy}/{MM}/{dd}/{slug}",
        },
      },
    },
  },
}
`;
